patches/server/0575-Configurable-door-breaking-difficulty.patch

```diff
@@ -5,16 +5,32 @@ Subject: [PATCH] Configurable door breaking difficulty
 
 
 diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
-index b289f6eb973d1ddf6cf913a0e995f899b9ecd041..e8b3ce26f289b961e2763be3b033c611bdfe583b 100644
+index aadfd01707b04a50daf846ef70c66f3f23108ff3..79fee47102f169a32cfb0635e548185a0359bcb0 100644
 --- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
 +++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
-@@ -109,6 +109,25 @@ public class PaperWorldConfig {
+@@ -109,6 +109,41 @@ public class PaperWorldConfig {
          disableMobSpawnerSpawnEggTransformation = getBoolean("game-mechanics.disable-mob-spawner-spawn-egg-transformation", disableMobSpawnerSpawnEggTransformation);
      }
  
++    public List<net.minecraft.world.Difficulty> zombieVillagerBreakDoors;
++    public List<net.minecraft.world.Difficulty> zombiePiglingBreakDoors;
 +    public List<net.minecraft.world.Difficulty> zombieBreakDoors;
 +    public List<net.minecraft.world.Difficulty> vindicatorBreakDoors;
 +    private void setupEntityBreakingDoors() {
```

Comment:
This should be redone into a map containing the 5 (not 4) entity types. Zombie, Zombie Villager, Zombified Piglin, Husk, and Vindicator.

Reply:
Maybe like this? https://gist.github.com/Doc94/9e9485891544da1bfae29009c0c38a38
i dont find a nice form for autogenerate the map with all zombie varians and vindicator (most if EntityType are not a enum)

Reply:
no, not really. you probably should use a ReferenceToObjectOpenHashMap that is initialized with the 5 EntityType keys.
nms EntityType is a pseudo-enum so using == to compare is fine (hence the "Reference")
then you can iterate over the keys and set the values since they all take the same form.

Reply:
hmm i dont understand about the ReferenceToObjectOpenHashMap if you have a example for this :/
i try make another form https://gist.github.com/Doc94/9e9485891544da1bfae29009c0c38a38 less code i think... but has a issue with the get of original values :/

Reply:
Presumably a setup that follows https://pastebin.com/azaRNmLz might work out for you :+1:

Reply:
Thanks lynx, i make a new commit with changes to the current system based in the comments :3